tests/testthat/test-format-report.R

```diff
@@ -131,6 +131,28 @@ describe("formatting functions", {
       unique(unname(unlist(exported_func_flex$footer$dataset))),
       "Testing directories: tests/testthat"
     )
+
+    ## Edge cases ##
+
+    # Test multiple testing directories
+    exports_df2 <- bind_rows(exports_df, exports_df)
+    exports_df2$test_dirs[[2]] <- "inst/unit-tests/tests"
+    exported_func_flex <- format_traceability_matrix(exports_df2, return_vals = TRUE)
+    expect_equal(
+      unique(unname(unlist(exported_func_flex$footer$dataset))),
+      "Testing directories: tests/testthat, inst/unit-tests/tests"
+    )
+
+    # No tests found for some functions
+    exports_df2$test_files[[2]] <- ""
+    exports_df2$test_dirs[[2]] <- NULL
```

Comment:
As far as I can tell, this regression test isn't working as intended.  If I revert the `format_traceability_matrix` code changes from f8f2ab1, the test still passes.

With this `NULL` assignment, I'm guessing your goal is to actually add `NULL` as an element, but it will instead drop the last element and recycle the first:

``` r
(d <- tibble::tibble(a = list("a1", "a2")))
#> # A tibble: 2 × 1
#>   a
#>   <list>
#> 1 <chr [1]>
#> 2 <chr [1]>
d$a
#> [[1]]
#> [1] "a1"
#>
#> [[2]]
#> [1] "a2"
d$a[[2]] <- NULL
d
#> # A tibble: 2 × 1
#>   a
#>   <list>
#> 1 <chr [1]>
#> 2 <chr [1]>
d$a
#> [[1]]
#> [1] "a1"
#>
#> [[2]]
#> [1] "a1"
```

You can instead do this:

``` r
d <- tibble::tibble(a = list("a1", "a2"))
d$a[2] <- list(NULL)
d
#> # A tibble: 2 × 1
#>   a
#>   <list>
#> 1 <chr [1]>
#> 2 <NULL>
d$a
#> [[1]]
#> [1] "a1"
#>
#> [[2]]
#> NULL
```

Reply:
Goot catch, cant believe I didn't notice that. Ironically I added this test to capture the very mistake I made within the function itself (mishandling of `NULL` elements in a table)